### PR TITLE
Story/cord 11

### PR DESCRIPTION
**Pull request by aa-kashk**
On the paper page, when hovering over a found location in the text the wikipedia title should be shown

### CODE REVIEW

**Comment by aa-kashk** [comment]
I was able to wrap the element correctly but the url issue(opening the same wiki title for all links) still persists. I checked the javascript code generated in the page source and looks like the final script is replacing all the hrefs in the previous paragraphs. I tried multiple things to fix this but I am unable to. 

One fix involved using the attr function to attach the url but although the url issue gets solved, the entire paragraph gets wrapped in the anchor again.

File: cord19-explorer/src/main/webapp/WEB-INF/views/paper.html [hunk 1]
```diff
@@ -66,7 +66,11 @@ <h3 th:text="*{section.title != ''} ? *{section.title}: 'No Section title'"></h3
 	      $(function() {
 	      	[# th:each="match : ${para.locationMatches}" ]
 	      	[# th:if="${match != null && match.selectedArticle != null}" ]
-	      	$("#[(${#ids.next('para')})]").mark("[(${match.locationName})]", { "separateWordSearch": false})	
+	      	$("#[(${#ids.next('para')})]").mark("[(${match.locationName})]", { "separateWordSearch": false})
+	     	var title = [[${match.selectedArticle.title}]]
+	      	$("#[(${#ids.next('para')})]").mark("[(${match.locationName})]", { "separateWordSearch": false}).attr('title', title);
+	        //$("#[(${#ids.next('para')})]").mark("[(${match.locationName})]", { "separateWordSearch": false}).wrap( "<a href='https://en.wikipedia.org/wiki/"+title+"' target=_blank></a>" );
+	        $( "mark" ).wrap( "<a href='https://en.wikipedia.org/wiki/"+title+"' target=_blank></a>" );
```
Comment (jdamerow) [inline cord19-explorer/src/main/webapp/WEB-INF/views/paper.html]: have you tried adding a span with a class that contains for example the index of the loop and use javascript to generate the link on click. something like this:
```
$("#[(${#ids.next('para')})]").mark("[(${match.locationName})]", { "separateWordSearch": false, "className": "wiki-[(${stats.index})]"})
$("mark.wiki-[(${stats.index})]).click(...)
```
does this make sense? this is about iterations in thymeleaf: https://www.baeldung.com/thymeleaf-iteration, and this is the mark.js documentation: https://markjs.io/

**Review by jdamerow — CHANGES_REQUESTED** [review]

**Comment by jdamerow** [comment]
Jenkins, test this please.

File: cord19-explorer/src/main/webapp/WEB-INF/views/paper.html [hunk 2]
```diff
@@ -64,9 +64,13 @@ <h3 th:text="*{section.title != ''} ? *{section.title}: 'No Section title'"></h3
...
 	      <script th:inline="javascript" th:if="${not #lists.isEmpty(para.locationMatches)}">
 	      $(function() {
-	      	[# th:each="match : ${para.locationMatches}" ]
+	      	[# th:each="match, iStat : ${para.locationMatches}" ]
 	      	[# th:if="${match != null && match.selectedArticle != null}" ]
-	      	$("#[(${#ids.next('para')})]").mark("[(${match.locationName})]", { "separateWordSearch": false})	
+	      	$("#[(${#ids.next('para')})]").mark("[(${match.locationName})]", { "separateWordSearch": false})
+	     	var title = [[${match.selectedArticle.title}]]
+	      	$("#[(${#ids.next('para')})]").mark("[(${match.locationName})]", { "separateWordSearch": false}).attr('title', title);
+	        $("#[(${#ids.next('para')})]").mark("[(${match.locationName})]", { "separateWordSearch": false, "className": "wiki-[(${#ids.next('para')})]-[(${iStat.index})]"})
```
Comment (jdamerow) [inline cord19-explorer/src/main/webapp/WEB-INF/views/paper.html:70]: you are marking the text 3 times resulting in 3 divs around the text. One should be enough

**Review by jdamerow — CHANGES_REQUESTED** [review]

**Comment by jdamerow** [comment]
Jenkins, test this please.

**Pull request merged**